app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -735,6 +736,14 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 }
                 return true;
 
+            case R.id.action_duplicate:
+                loyaltyCard = DBHelper.getLoyaltyCard(database, loyaltyCardId);
+                DBHelper.insertLoyaltyCard(database,loyaltyCard.store,loyaltyCard.note,loyaltyCard.expiry,loyaltyCard.balance,loyaltyCard.balanceType,
+                        loyaltyCard.cardId,loyaltyCard.barcodeId,loyaltyCard.barcodeType,loyaltyCard.headerColor,loyaltyCard.starStatus,loyaltyCard.lastUsed);
+                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                startActivity(intent);
```

Comment:
Issue #731 talks about starting the create activity with all values pre-filled, which I think would be a better workflow because you could change things before saving (I don't see a single use case in having 2 exact duplicate cards).

I think the cleanest way to do that would probably be to extend the `extractIntentFields` function in `LoyaltyCardEditActivity` to take a new field and if that's given to pre-fill everything (maybe call it `duplicateFromLoyaltyCardId`?).

I suppose basing this on the ID of the original LoyaltyCard in the database would be easiest, as that way you could also already load in the photos (your duplication code currently doesn't duplicate photos).

Reply:
Well, i think that this is what you asked for but if something is missing please let me know.

Reply:
Well, right now you just changed the code to call the LoyaltyCardEditActivity and then instantly call the MainActivity from there.

I more meant like: open the LoyaltyCardEditActivity with all the fields filled in already with the values of the card you're copying, so you can click "Save" to make a copy or make changes first.

Reply:
Oh, you're completely right. My bad, I didn't test to confirm and completely forgot that `updateLoyaltyCard` is the variable that also causes the card to be loaded from the database. So this part actually works perfectly well and it's the smallest code change possible :+1:

Reply:
Ok nice, glad to hear that :D